pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/mercury/ClinicalOrderDao.java

```diff
@@ -0,0 +1,113 @@
+package org.broadinstitute.dsm.db.dao.mercury;
+
+import static org.broadinstitute.ddp.db.TransactionWrapper.inTransaction;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Optional;
+
+import lombok.extern.slf4j.Slf4j;
+import org.broadinstitute.dsm.db.dao.Dao;
+import org.broadinstitute.dsm.db.dao.ddp.instance.DDPInstanceDao;
+import org.broadinstitute.dsm.db.dto.ddp.instance.DDPInstanceDto;
+import org.broadinstitute.dsm.db.dto.mercury.ClinicalOrderDto;
+import org.broadinstitute.dsm.db.dto.mercury.ClinicalOrderUseCase;
+import org.broadinstitute.dsm.statics.DBConstants;
+import org.broadinstitute.lddp.db.SimpleResult;
+
+@Slf4j
+public class ClinicalOrderDao implements Dao<ClinicalOrderDto> {
+
+    public static String SQL_GET_ALL_ORDERS_FOR_REALM = "select ms.order_id, ms.ddp_participant_id, "
+            + "IFNULL(t.collaborator_sample_id, kit.bsp_collaborator_sample_id) as collaborator_sample_id, order_date, order_status, "
+            + "status_date, status_detail,  IF(ms.tissue_id is null, \"Normal\", \"Tumor\") as sample_type "
+            + "FROM ddp_mercury_sequencing ms "
+            + "LEFT join ddp_tissue t on (ms.tissue_id = t.tissue_id  and ms.tissue_id ) "
+            + "LEFT join ddp_kit_request kit on (ms.dsm_kit_request_id = kit.dsm_kit_request_id)"
+            + "WHERE ms.ddp_instance_id = ? order by order_date desc";
+
+    public static String COMPLETED_ORDER_STATUS = "Completed";
+
+    @Override
+    public int create(ClinicalOrderDto clinicalOrderDto) {
+        return 0;
+    }
+
+    @Override
+    public int delete(int id) {
+        return 0;
+    }
+
+    @Override
+    public Optional<ClinicalOrderDto> get(long id) {
+        return Optional.empty();
+    }
+
+    public Collection<ClinicalOrderDto> getOrdersForRealm(String realm, String projectId, String topicId,
+                                                          ClinicalOrderUseCase clinicalOrderUseCase) {
+        DDPInstanceDto ddpInstance = new DDPInstanceDao().getDDPInstanceByInstanceName(realm).orElseThrow();
+        HashMap<String, ClinicalOrderDto> map = new HashMap<>();
+        ArrayList<ClinicalOrderDto> noStatusOrders = new ArrayList<>();
+        SimpleResult results = inTransaction((conn) -> {
+            SimpleResult dbVals = new SimpleResult();
+            try (PreparedStatement stmt = conn.prepareStatement(SQL_GET_ALL_ORDERS_FOR_REALM)) {
+                stmt.setInt(1, ddpInstance.getDdpInstanceId());
```

Comment:
I would create inner select like `WHERE ms.ddp_instance_id = (SELECT ddp_instance_id FROM ddp_instance WHERE instance_name = ?)`, this would help to eliminate line 54.